g11n-ws/vip-manager-l10n/src/main/java/com/vmware/l10n/source/service/impl/SyncI18nSourceServiceImpl.java

```diff
@@ -0,0 +1,174 @@
+
+import com.vmware.l10n.source.service.SyncI18nSourceService;
+import com.vmware.l10n.utils.DiskQueueUtils;
+import com.vmware.vip.api.rest.APIParamName;
+import com.vmware.vip.api.rest.APIV2;
+import com.vmware.vip.common.constants.ConstantsKeys;
+import com.vmware.vip.common.exceptions.VIPHttpException;
+import com.vmware.vip.common.http.HTTPRequester;
+import com.vmware.vip.common.l10n.exception.L10nAPIException;
+import com.vmware.vip.common.l10n.source.dto.ComponentSourceDTO;
+
+/**
+ * 
+ * This class used to send the collection source to singleton server
+ *
+ */
+
+@Service
+public class SyncI18nSourceServiceImpl implements SyncI18nSourceService {
+	
+	private final static Logger LOGGER = LoggerFactory.getLogger(SyncI18nSourceServiceImpl.class);
+	private final static String LOCAL_STR = "local";
+	private final static String I18N_STR = "i18n";
+
+	/** the path of local resource file,can be configured in spring config file **/
+	@Value("${source.bundle.file.basepath}")
+	private String basePath;
+
+	/** the url of Singleton API,can be configed in spring config file **/
+	@Value("${vip.server.url}")
+	private String remoteVIPURL;
+
+	@Value("${vip.server.authentication.enable}")
+	private boolean remoteVIPAuthEnable;
+
+	@Value("${vip.server.authentication.appId:#}")
+	private String remoteVIPAuthAppId;
+
+	@Value("${vip.server.authentication.token:#}")
+	private String remoteVIPAuthAppToken;
+
+	/**
+	 * it represents the remote singleton service is available.
+	 */
+	private boolean singletonConnected = false;
+
+	/**
+	 * 
+	 * Synchronize the updated source to remote Singleton server
+	 *
+	 */
+	@Override
+	public synchronized void sendSourceToI18n() {
+		if (LOCAL_STR.equalsIgnoreCase(remoteVIPURL)) {
+			return;
+		}
+		try {
+			pingSingleton(this.remoteVIPURL);
+			setSingletonConnected(true);
+			processSingletonQueueFiles();
+		} catch (L10nAPIException e) {
+			LOGGER.error("Remote [" + remoteVIPURL + "] is not connected.", e);
+			setSingletonConnected(false);
+		}catch (Exception e) {
+			LOGGER.error(e.getMessage(), e);
+		}
+	}
+
+	/**
+     * scan the Singleton cached source file DIR and process the cached source file
+     */
+	private synchronized void processSingletonQueueFiles() {
+
+		List<File> queueFiles = DiskQueueUtils.listQueueFiles(new File(basePath + DiskQueueUtils.L10N_TMP_I18N_PATH));
+		if (queueFiles == null) {
+			return;
+		}
+		LOGGER.debug("the Singleton cache file size---{}", queueFiles.size());
+
+		for (File quefile : queueFiles) {
+			try {
+				Map<String, ComponentSourceDTO> mapObj = DiskQueueUtils.getQueueFile2Obj(quefile);
+				for (Entry<String, ComponentSourceDTO> entry : mapObj.entrySet()) {
+					ComponentSourceDTO cachedComDTO = entry.getValue();
+					sendData2RemoteVIP(cachedComDTO);
+				}
+				DiskQueueUtils.moveFile2IBackupPath(basePath, quefile, I18N_STR);
+			} catch (VIPHttpException e) {
+				LOGGER.error("Send source file to Singleton error:", e);
+				break;
```

Comment:
why need to break all queueFiles handling?

Reply:
because the network not available